pydantic/_internal/_decorators.py

```diff
@@ -489,22 +489,36 @@ def unwrap_wrapped_function(
     Returns:
         The underlying function of the wrapped function.
     """
-    all: tuple[Any, ...]
+    all: list[Any]
+    try:
+        from functools import cached_property
+
+        cached_property_list = [cached_property]
+    except ImportError:
+        cached_property = int  # anything that doesn't match isinstance below
+        cached_property_list = []
+
     if unwrap_class_static_method:
-        all = (
+        all = [
             staticmethod,
             classmethod,
             partial,
             partialmethod,
-        )
+            property,
+            *cached_property_list,
+        ]
     else:
-        all = partial, partialmethod
+        all = [partial, partialmethod, property, *cached_property_list]
 
-    while isinstance(func, all):
+    while isinstance(func, tuple(all)):
```

Comment:
another super minor thing — any reason not to make `all` a tuple in the first place?

Reply:
Just that type checkers are less happy:
<img width="760" alt="image" src="https://github.com/pydantic/pydantic/assets/1755071/ae789a97-0acb-4d3d-bf36-9a003ad98021">
No idea why `isinstance` _requires_ a tuple, I suspect that is conceptually wrong.